### ISSUE TITLE
Relative orders emergency break

### ISSUE THREAD

**Issue by CryptoKong** [Codaone/DEXBot#231]
Normally relative orders tool calculates the centre price from two closest order then places orders at desired spread. Once an order is filled it re calculates centre price and places 2 new orders at desired spread.

I would like to suggest that once an order is filled the user should be able to set a limit on the newly calculated centre price.

For example: Sometimes a large chunk of the order book can be filled and when the worker calculates the new centre price it can be a long way from where it was previously. This can often result in bad orders being placed at a large offset.

If the user could tell the worker to only reset the orders if the new centre price is within X% of the old centre price or if there was a minimum spread of Y% on the market, it could be very helpful in handling massive unexpected pump and dumps whilst the worker only resumes placing orders when certain market conditions return.

**Comment by the-scott-davis** [comment]
Agree. +1

**Comment by edwinux** [comment]
+1

**Comment by MarkoPaasila** [comment]
Would a moving average solve this?

**Comment by CryptoKong** [comment]
Im not sure if a moving average would solve it... if the price moves away from the average you would ideally like the worker to stop but then resume when conditions were improved which might not be reversion back to the mean. 

IMO it would be best to restart when a chosen amount of liquidity is back on the books. Ie: a minimum required spread to calculate a centre price from.

Or a possible combination of the above and a check that new centre price is within given range from last centre price.

If your trying to work a flat market with a tight spread. A sharp move away from the last centre price would not allow the worker to replace orders again until say 3 consectutive centre prices within a given range. Perhaps the number of consecutive centre prices within range could also be variable.

**Comment by the-scott-davis** [comment]
Yep. Statistically, there is no full-proof way to handle this without some obvious variance in performance. I think a general consensus on the best way to handle it should cover the majority of the scenarios though.

**Comment by MarkoPaasila** [comment]
We need to improve markets using staggered orders first, then this should be no issue any more.

**Comment by thehapax** [comment]
@MarkoPaasila still on hold or should we re-open for development?

**Comment by MarkoPaasila** [comment]
I think the original suggestion by Kong could be useful. It shouldn't be hard, should it?

**Comment by bitfag** [comment]
The implementation should be relatively easy and the feature make sense if trading a volatile market. Just stop all trades if price moved too far.

**Comment by MarkoPaasila** [comment]
How about _emergency break_ if n orders on one side are filled and none on the other? That seems like a clear indication of a change in price, without having to store price data or making assumptions.

When to end the break is another question. Is our only option to measure volatility and set thresholds? Any ideas?
